workstreams/events_in_cicd/vocabulary.md

```diff
@@ -0,0 +1,108 @@
+- **Event Type**: defines a textual description of the event type, only event types described in this document are supported. All event types should be prefixed with `CD.`
+- **Event Source**: defines the context in which an event happened
+- **Event Timestamp**: defines the time when the event was produced
+
+The following sections list the events in different phases, allowing adopters to choose the events that they are more interested in.
+
+# Source Code Version Control Events
+
+This phase includes events related to changes in Source Code repositories that are relevant from a Continuous Delivery perspective.
+
+
+## Repository Events
+
+These events are related to Source Code repositories
+- **Repository Created Event**: a new Source Code Repository was created to host source code for a project
+- **Repository Modified Event**: a Source Code Repository modified some of its attributes, like location, or owner
+- **Repository Deleted Event**: a Source Code Repository was deleted and it is not longer available
+- **Branch Created Event**: a Branch inside the Repository was created 
+- **Branch Deleted Event**: a Branch inside the Repository was deleted
+- **Change Committed**: a change is commited to a branch
+- **Change Approved**: a change request is created by a user, that requires review and approval before being applied to an existing branch inside the repository
+- **Change Rejected**: a change was reviewed and rejected by a user
+- **Change Merged**: a change request has been merged to a branch in an existing repository
+
+
+Repository Events MUST include the following attributes:
+- **Event Type**: the type is restricted to include `CD.Repository**` prefix. For example `CD.Repository.Created` or `CD.Repository.ChangeApproved`
+- **Repository URL**: indicates the location of the source code repository
+- **Repository Name**: friendly name to list this repository to users
+
+Optional attributes: 
+- **Repository Owner**: indicates who is the owner of the repository, usually a `user` or an `organization`
+
+
+# Continuous Integration Pipeline Events
+
+These events are related to continuous integration pipelines, this pipelines usually include building, testing, packaging and releasing software artifacts. 
+Due the dynamic nature of Pipelines, most of actual work needs to be queued to happen in a distributed way, hence Queued events are added. 
+Adopters can choose to ignore these events if they don't apply to their use cases. 
+
+A pipeline, in the context of Continuous Integration, is the definition of a set of tasks that needs to be performed to build, test, package and release software artifacts. A pipeline can be instanciated multiple times, for example to build different versions of the same artifact. That instance will have a unique Id and it will help us to track the build and release progress on a particular software artifact. 
+
+- **PipelineInstance Queued**: a PipelineInstance has been schedule to run
```

Comment:
In ESCI, a pipeline run (ESCI Flow Run) cannot be queued, only activity runs in the pipeline run can be. The pipeline run could be considered "Queued" if the first activity run in the pipeline is queued.

But, as mentioned above, these events can be ignored if they don't apply.

Reply:
@erkist I think that is the intention if your system supports queueing we have the event to cover that, if it doesn't support it then the event is not needed. I think that systems such as Travis CI, Jenkins does support Queueing of Pipeline Instances/Runs